client/log_client.go

```diff
@@ -255,25 +260,25 @@ func (c *LogClient) GetAndVerifyInclusionAtIndex(ctx context.Context, data []byt
 	return c.VerifyInclusionAtIndex(root, data, index, resp.Proof.Hashes)
 }
 
-func (c *LogClient) getAndVerifyInclusionProof(ctx context.Context, leafHash []byte, sth *trillian.SignedLogRoot) error {
+func (c *LogClient) getAndVerifyInclusionProof(ctx context.Context, leafHash []byte, sth *trillian.SignedLogRoot) (bool, error) {
 	resp, err := c.client.GetInclusionProofByHash(ctx,
 		&trillian.GetInclusionProofByHashRequest{
 			LogId:    c.LogID,
 			LeafHash: leafHash,
 			TreeSize: sth.TreeSize,
 		})
 	if err != nil {
-		return err
+		return false, err
 	}
 	if len(resp.Proof) < 1 {
-		return errors.New("no inclusion proof supplied")
+		return false, nil
```

Comment:
You seem to have moved the creation of an error for this condition up a level to the caller (L239 is effectively `if err == nil && !ok`, why not just return `false, errors.New("no inclusion...")` as before?

Reply:
An error here breaks WaitForInclusion().